pkg/rpcserver/rpcserver.go

```diff
@@ -117,20 +151,77 @@ func startRPCServer(mgr *Manager) (*RPCServer, error) {
 			"Number of times fuzzer was stalled with no exec requests", stats.Rate{}),
 		statNoExecDuration: stats.Create("no exec duration",
 			"Total duration fuzzer was stalled with no exec requests (ns/sec)", stats.Rate{}),
-		statCoverFiltered: stats.Create("filtered coverage", "", stats.NoGraph),
 	}
-	s, err := flatrpc.ListenAndServe(mgr.cfg.RPC, serv.handleConn)
+	s, err := flatrpc.ListenAndServe(cfg.RPC, serv.handleConn)
 	if err != nil {
 		return nil, err
 	}
-	baseSource.Store(serv.checker)
-
-	log.Logf(0, "serving rpc on tcp://%v", s.Addr)
-	serv.port = s.Addr.Port
+	serv.serv = s
+	serv.Port = s.Addr.Port
 	return serv, nil
 }
 
-func (serv *RPCServer) handleConn(conn *flatrpc.Conn) {
+func (serv *Server) Close() error {
+	return serv.serv.Close()
+}
+
+type VMState struct {
+	State     int
+	Timestamp time.Time
+}
+
+const (
+	StateOffline = iota
+	StateBooting
+	StateFuzzing
+	StateStopping
+)
+
+func (serv *Server) VMState() map[string]VMState {
+	serv.mu.Lock()
+	defer serv.mu.Unlock()
+	return maps.Clone(serv.info)
+}
+
+func (serv *Server) MachineInfo(name string) []byte {
+	serv.mu.Lock()
+	runner := serv.runners[name]
+	if runner.conn == nil || runner.stopped {
```

Comment:
It will panic if the runner does not exist, right?

Reply:
Fixed.